### PR TITLE
Add android client to the repo

### PR DESCRIPTION
**Pull request by pratikvelani**
- Add android client to the repo

### CODE REVIEW

**Review by francesca64 — APPROVED** [review]

**Comment by jamilbk** [comment]
@pratikvelani ~~Could you [set up commit signing for Git](https://docs.github.com/en/authentication/managing-commit-signature-verification/signing-commits)? I'll override for now but our branch protection rules require them.~~

Screenshot 2023-07-06 at 6 19 37 AM

EDIT: Nevermind, I see you've set it up!

File: kotlin/android/LICENSE [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+      To apply the Apache License to your work, attach the following
+      boilerplate notice, with the fields enclosed by brackets "[]"
+      replaced with your own identifying information. (Don't include
+      the brackets!)  The text should be enclosed in the appropriate
+      comment syntax for the file format. We also recommend that a
+      file or class name and description of purpose be included on the
+      same "printed page" as the copyright notice for easier
+      identification within third-party archives.
+
+   Copyright [yyyy] [name of copyright owner]
```
Comment (jamilbk) [inline kotlin/android/LICENSE]:
```suggestion
   Copyright 2023 Firezone, Inc.
```

File: kotlin/android/app/src/main/res/font/font.xml [hunk 2]
```diff
@@ -0,0 +1,9 @@
+<?xml version="1.0" encoding="utf-8"?>
+<font-family xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:tools="http://schemas.android.com/tools">
+    <font
```
Comment (jamilbk) [inline kotlin/android/app/src/main/res/font/font.xml:5]: Ah, we decided to go with `Source Sans Pro` for 1.0. I think a FIXME here is ok.

```suggestion
    <!-- FIXME: Use Source Sans Pro for 1.0 -->
    <font
```

File: kotlin/android/app/src/main/res/mipmap-hdpi/ic_launcher.png
Comment (jamilbk) [inline kotlin/android/app/src/main/res/mipmap-hdpi/ic_launcher.png:1]: Wow, thanks for adding all these assets! We'll probably want to double-check the background color as we near launch.

File: kotlin/android/settings.gradle [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+Properties properties = new Properties()
+properties.load(new FileInputStream(new File('local.properties')))
+
+dependencyResolutionManagement {
+    repositoriesMode.set(RepositoriesMode.FAIL_ON_PROJECT_REPOS)
+    repositories {
+        google()
+        maven {
+            url = uri('https://maven.pkg.github.com/firezone/connlib')
```
Comment (jamilbk) [inline kotlin/android/settings.gradle:17]: Isn't this a path dependency now?

**Review by jamilbk — CHANGES_REQUESTED** [review]
Just missing a CI file similar to the one @roop added [here](https://github.com/firezone/firezone/pull/1737/files#r1254386100) but let's table that for another PR.

Also, I believe we're ditching the binary / package dependency for `Connlib` over a path-based one that builds alongside the main project. We can save that for another PR too.

**Comment by jamilbk** [comment]
@pratikvelani Looks like there are some linter issues, [one with mixed line endings](https://github.com/firezone/firezone/actions/runs/5475978599/jobs/9972839408?pr=1738#step:6:22) and other about whitespace. I think the whitespace one we can ignore for that directory if Android Studio is consistently violating the linter here.

File: kotlin/android/settings.gradle (hunk 3 above)
Comment (jamilbk) [reply kotlin/android/settings.gradle:17]: Edit: we can save this for another PR: https://github.com/firezone/product/issues/626

**Review by jamilbk — APPROVED** [review]

**Comment by pratikvelani** [comment]
> @pratikvelani ~Could you [set up commit signing for Git](https://docs.github.com/en/authentication/managing-commit-signature-verification/signing-commits)? I'll override for now but our branch protection rules require them.~
> 
> Screenshot 2023-07-06 at 6 19 37 AM
> EDIT: Nevermind, I see you've set it up!

Yupe, I did the first commit from my MacBook it didnt have the gpg keys setup.

**Pull request merged**